xmtp_db/src/encrypted_store/database/native/sqlcipher_connection.rs

```diff
@@ -270,7 +270,16 @@ impl super::ValidatedConnection for EncryptedConnection {
             SELECT count(*) FROM sqlite_master;",
             self.pragmas()
         ))
-        .map_err(|_| PlatformStorageError::SqlCipherKeyIncorrect)?;
+        .map_err(|e| {
+            let err_msg = e.to_string();
+            if err_msg.contains("database is locked") {
```

Comment:
does the lock error happen only when the connection is being validated? I would think it happens on a read/write or maybe when its being acquired